NUsenseCpp/Core/Src/platform/NUsense/NUsenseIO/loop.cpp

```diff
@@ -98,22 +110,31 @@ namespace platform::NUsense {
                         break;
                 }
             }
+
+            // If we are cooling down, then see whether the timer has timed out. If so, then send
+            // the next write-instruction.
+            if ((status_states[(uint8_t) current_id - 1] == WRITE_1_COOLDOWN)
+                && (torque_cooldown_timers[i].has_timed_out())) {
+
+                send_servo_write_2_request(current_id, i);
+                status_states[(uint8_t) current_id - 1] = WRITE_2_RESPONSE;
+            }
         }
 
         // Handle the incoming protobuf messages from the nuc.
         if (nuc.handle_incoming()) {
             // For every new target, update the state if it is a servo.
             message_actuation_ServoTargets* new_targets = nuc.get_targets();
             for (int i = 0; i < new_targets->targets_count; i++) {
-                auto new_target = new_targets->targets[i];
-                if ((new_target.id) < NUMBER_OF_DEVICES) {
-                    servo_states[new_target.id].profile_velocity =
-                        (float(new_target.time.seconds) * 1000) + (float(new_target.time.nanos) / 1e6);
-                    servo_states[new_target.id].position_p_gain  = new_target.gain;
-                    servo_states[new_target.id].goal_position    = new_target.position;
-                    servo_states[new_target.id].torque           = new_target.torque;
+                auto new_target = &(new_targets->targets[i]);
```

Comment:
```suggestion
                message_actuation_ServoTarget new_target = &(new_targets->targets[i]);
```

Better to be more explicit with the type when it is not known right away, e.g.,

`auto sensors = std::make_unique<Sensors>();` is fine since the type sensors is there. Just for the sake of readability

Reply:
Done, although it has to be `message_actuation_ServoTarget*`.